src/providers/files/files_ops.c

```diff
@@ -829,15 +1008,26 @@ static void sf_enum_files_first_step(struct tevent_context *ev,
                       filename);
             } else if (ret != EOK) {
                 DEBUG(SSSDBG_OP_FAILURE,
-                      "Cannot enumerate groups from %s, aborting\n",
+                      "Cannot enumerate users from %s, aborting\n",
                       filename);
                 goto done;
             }
         } else {
+            id_ctx->refresh_ctx->updating_passwd = REFRESH_NOT_RUNNIG;
             if (state->flags & SF_UPDATE_GROUP) {
-                state->current_step = DELETE_GROUPS;
+                state->current_step = WAIT_TO_START_GROUPS;
             } else {
-                state->current_step = UPDATE_FINISH;
+                if (state->id_ctx->refresh_ctx->passwd_start_again) {
+                    state->id_ctx->refresh_ctx->passwd_start_again = false;
+                    id_ctx->refresh_ctx->updating_passwd = REFRESH_WAITING_TO_START;
+                    state->current_step = WAIT_TO_START_USERS;
+                } else if (state->id_ctx->refresh_ctx->group_start_again) {
+                    state->id_ctx->refresh_ctx->group_start_again = false;
+                    id_ctx->refresh_ctx->updating_groups = REFRESH_WAITING_TO_START;
+                    state->current_step = WAIT_TO_START_GROUPS;
```

Comment:
This looks a bit odd taking into account `(state->flags & SF_UPDATE_GROUP)` isn't set here.
Or do I miss something...

Reply:
Hi,

`state->flags` only contain the information from the initial call to `sf_enum_files_send()`, if the request is already running and new changes happen this is recorded in the `*_start_again` variables.

HTH

bye,
Sumit